### PR TITLE
Pressure equalizer documentation

### PR DESCRIPTION
**Pull request by igiannakas**
A starting point for pressure equalizer documentation / extrusion rate smoothing for your review.

Have not run a spell checker on it and a second eye on the equations would be excellent :) 

@Noisyfox  @SoftFever would appreciate your inputs as well as the theory is at the boundary of my 3d printing and firmware knowledge :)

### CODE REVIEW

**Comment by igiannakas** [comment]
Specifically what I need to clarify is whether the firmware lookahead planners start slowing down the print speed ahead of a slower feature (eg an overhang or a bridge) or after the slower feature starts printing.

**Comment by SoftFever** [comment]
Wow, Impressive!
That's serious document! 

Give me some time to review it.

**Comment by igiannakas** [comment]
Thought more detail is better than less haha! Also have been trying to tune this for my printer and trying to figure out what a good value may be 🤣

File: doc/Extrusion-rate-smoothing.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+<h1>Extrusion rate smoothing (known as pressure equalizer in Prusa Slicer)</h1>
```
Comment (SoftFever) [inline doc/Extrusion-rate-smoothing.md:1]: I'm thinking would it be better to move `(known as pressure equalizer in Prusa Slicer)` to summary paragraph below?

File: doc/Extrusion-rate-smoothing.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+<h2>Limitations:</h2>
+**This feature can only work where speed changes are induced by the slicer** - for example when transitioning from fast to slow print moves when printing overhangs, bridges and from printing internal features to external features and vice versa. 
+
+However, it will not affect extruder behaviour when the printer is slowing down due to firmware commands - for example when turning around corners. 
+
+In this case, the printer slows down and then accelerates independently of what the slicer has requested. In this case, the slicer is commanding a consistent speed; however, the printer is adjusting this to operate within its printer kinematic limits (SCV/Jerk) and accelerations. As the slicer is not aware of this slow down, it cannot apply pre-emptive extrusion rate smoothing to the feature and instead, the changes are governed by the printer firmware exclusively.
```
Comment (SoftFever) [inline doc/Extrusion-rate-smoothing.md:154]: Could you add a credits section at the end of the page, add put your name there?

**Comment by SoftFever** [comment]
> Thought more detail is better than less haha! Also have been trying to tune this for my printer and trying to figure out what a good value may be 🤣

Haha, it's a shining example of how the Orca WIKI should be written.

**Comment by SoftFever** [comment]
This WIKI is FANTASTIC! 
The technical details align perfectly with my understanding. 
You've explained everything exceptionally well.

File: doc/Extrusion-rate-smoothing.md (hunk 2 above)
Comment (igiannakas) [reply doc/Extrusion-rate-smoothing.md:164]: Done, added credits for the feature and its evolution plus docs

File: src/slic3r/GUI/Tab.cpp [hunk 3]
```diff
@@ -2156,8 +2156,8 @@ void TabPrint::build()
         optgroup->append_single_option_line("travel_jerk");
         
         optgroup = page->new_optgroup(L("Advanced"), L"param_advanced", 15);
-        optgroup->append_single_option_line("max_volumetric_extrusion_rate_slope");
-        optgroup->append_single_option_line("max_volumetric_extrusion_rate_slope_segment_length");
+        optgroup->append_single_option_line("max_volumetric_extrusion_rate_slope", "extrusion-rate-smoothing");
```
Comment (SoftFever) [inline src/slic3r/GUI/Tab.cpp:2159]: @igiannakas

**Comment by SoftFever** [comment]
Awesome.
I associated the option to this coming WIKI page.
Users will be able to click to option in Orca to open this WIKI in browser.
image

**Review by SoftFever — APPROVED** [review]

File: src/slic3r/GUI/Tab.cpp (hunk 3 above)
Comment (igiannakas) [reply src/slic3r/GUI/Tab.cpp:2159]: Ah that’s how you do it! Perfect thank you, will know for next time! 👌🏻

**Pull request merged**